src/ol/source/ImageWMS.js

```diff
@@ -191,6 +191,41 @@ class ImageWMS extends ImageSource {
       1, sourceProjectionObj || projectionObj, baseParams);
   }
 
+  /**
+   * Return the GetLegendGraphic URL for the passed resolution.
+   * Return `undefined` if the GetLegendGraphic URL cannot be constructed.
+   * @param {number} resolution Resolution.
+   * @param {!Object} params GetLegendGraphic params. Default `FORMAT` is
+   *     `image/png`. `VERSION` should not be specified here.
+   * @return {string|undefined} GetLegendGraphic URL.
+   * @api
+   */
+  getGetLegendGraphicUrl(resolution, params) {
+    const layers = this.params_.LAYERS;
+    const isSingleLayer = !Array.isArray(layers) || this.params_['LAYERS'].length === 1;
+    if (this.url_ === undefined || !isSingleLayer) {
+      return undefined;
+    }
+
+    const units = this.getProjection() && this.getProjection().getUnits();
+    const dpi = 25.4 / 0.28;
+    const mpu = METERS_PER_UNIT[units || 'm'];
+    const inchesPerMeter = 39.37;
+    const scale = resolution * mpu * inchesPerMeter * dpi;
```

Comment:
I'm not sure about the transformation of `resolution` to scale.
Couldn't find a code block where this was solved already and i think the pixelratio might be relevant here, too.
Advice appreciated.

Reply:
Thx. I adapted the code a little to use `getMetersPerUnit`.

Reply:
But there is a problem: you actually need the projection of the view not the projection of the source as the resolution is tied to the view projection.

Reply:
Oh of course. Hmm. Maybe i should change the parameters to pass a view instead of a resolution.
This way i can get the resolution and view-projection at once.

Reply:
I f you compare with https://github.com/openlayers/openlayers/blob/master/src/ol/source/ImageWMS.js#L159 you see that normally you would pass the resolution and projection as independent parameters. I think that is for greater usability in a context where you might have the needed parameters but not the view and more general less dependencies.

Reply:
also scale is not a mandatory parameter for legendgraphic requests. I don't know if it should be possible to call the method without these parameters? What do you think?